docs/architecture/blockchain.md

```diff
@@ -0,0 +1,70 @@
+# Blockchain
+
+The Miden blockchain protocol describes how the [state](state.md) progresses. In Miden, `Block`s are containers that aggregate account state changes and their proofs, together with created and consumed notes. Each `Block` is accompanied by a corresponding proof that attests to the correctness of all state transitions it contains.
+
+`Block`s represent the delta of the global [state](state.md) between two time periods. The current global state can be derived by applying all the blocks to the genesis state.
+
+Miden's blockchain protocol aims for the following:
+
+- **Proven transactions**: All included transactions have already been proven and verified when they reach the block.
+- **Fast genesis syncing**: New nodes efficiently sync to the network through a multi-step process:
+
+  1. Download historical `Block`s from genesis to the present.
+  2. Verify zero-knowledge proofs for all `Block`s.
+  3. Retrieve current state data (accounts, notes, and nullifiers).
+  4. Validate that the downloaded state matches the latest `Block`'s state commitment.
+
+This approach enables near-instant blockchain syncing by verifying `Block` proofs rather than re-executing individual transactions, resulting in exponentially faster performance. Consequently, state sync is dominated by the time needed to download the data.
```

Comment:
"near-instant" is probably stretching it. With the current process we still need to download all block headers + proofs and if the chain is long, this would definitely be minutes or maybe even hours.